cmd/main.go

```diff
@@ -338,33 +364,39 @@ func run(mainConfig Config) (err error) {
 	}
 	//+kubebuilder:scaffold:builder
 
-	if err = (&controller.PodReconciler{
-		Client:           mgr.GetClient(),
-		Scheme:           mgr.GetScheme(),
-		Triggerer:        entryReconciler,
-		IgnoreNamespaces: mainConfig.ignoreNamespacesRegex,
-	}).SetupWithManager(ctx, mgr); err != nil {
-		setupLog.Error(err, "unable to create controller", "controller", "Pod")
-		return err
-	}
-	if err = (&controller.EndpointsReconciler{
-		Client:           mgr.GetClient(),
-		Scheme:           mgr.GetScheme(),
-		Triggerer:        entryReconciler,
-		IgnoreNamespaces: mainConfig.ignoreNamespacesRegex,
-	}).SetupWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create controller", "controller", "Endpoints")
-		return err
+	if mainConfig.reconcile.ClusterSPIFFEIDs {
+		if err = (&controller.PodReconciler{
+			Client:           mgr.GetClient(),
+			Scheme:           mgr.GetScheme(),
+			Triggerer:        entryReconciler,
+			IgnoreNamespaces: mainConfig.ignoreNamespacesRegex,
+		}).SetupWithManager(ctx, mgr); err != nil {
+			setupLog.Error(err, "unable to create controller", "controller", "Pod")
+			return err
+		}
+		if err = (&controller.EndpointsReconciler{
+			Client:           mgr.GetClient(),
+			Scheme:           mgr.GetScheme(),
+			Triggerer:        entryReconciler,
+			IgnoreNamespaces: mainConfig.ignoreNamespacesRegex,
+		}).SetupWithManager(mgr); err != nil {
+			setupLog.Error(err, "unable to create controller", "controller", "Endpoints")
+			return err
+		}
 	}
 
-	if err = mgr.Add(manager.RunnableFunc(entryReconciler.Run)); err != nil {
-		setupLog.Error(err, "unable to manage entry reconciler")
-		return err
+	if mainConfig.reconcile.ClusterSPIFFEIDs || mainConfig.reconcile.ClusterStaticEntries {
```

Comment:
Probably more robust to just check if entryReconciler is != nil, implying something needed it so it was initialized, and therefore should be run.
```suggestion
	if entryReconciler != nil {
```